velox/functions/prestosql/benchmarks/ArraySumBenchmark.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+#include "velox/functions/Macros.h"
+#include "velox/functions/Registerer.h"
+#include "velox/functions/lib/benchmarks/FunctionBenchmarkBase.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+
+namespace {
+
+template <typename IT, typename OT>
+VELOX_UDF_BEGIN(array_sum)
+FOLLY_ALWAYS_INLINE bool call(OT& out, const arg_type<Array<IT>>& array) {
```

Comment:
could we also try to use the call null free fast path to remove some of these checks?

> https://facebookincubator.github.io/velox/develop/scalar-functions.html#null-free-fast-path

CC: @kevinwilfong

Reply:
Done.